lib/ClangImporter/ImportType.cpp

```diff
@@ -2301,8 +2301,11 @@ ClangImporter::Implementation::importParameterType(
     }
 
     paramTy = refType->getPointeeType();
-    if (!paramTy.isConstQualified())
+    if ((isa<clang::LValueReferenceType>(paramTy) &&
```

Comment:
So I found that checking for canonical allowed the certain methods that should remain `inout` to be import as such, and some overload methods to import with just value. 
```
    paramTy = refType->getPointeeType();
    if ((paramTy.isCanonical() && 
        !paramTy->isRValueReferenceType())) {
      isInOut = true;
    }
  }
```

For example.`HasAmbiguousMethods`  is now imported correctly:

```
struct HasAmbiguousMethods {
  init()
  func increment(_ a: Int32) -> Int32
  mutating func incrementMutating(_ a: Int32) -> Int32
  mutating func incrementMutating(_ a: Int32, _ b: Int32, _ c: inout Int32)
  func increment(_ a: Int32, _ b: Int32, _ c: inout Int32)
  mutating func incrementMutating(_ a: inout Int32, _ b: Int32)
  func increment(_ a: inout Int32, _ b: Int32)
  func numberOfMutableMethodsCalled() -> Int32
  mutating func numberOfMutableMethodsCalledMutating() -> Int32
}
struct HasAmbiguousMethods2 {
  init()
  func increment(_ a: Int32) -> Int32
}
```

And the `push_back` overloads are also correct:
```
      mutating func push_back(_ __x: std.__1.__CxxTemplateInstNSt3__16vectorIdNS_9allocatorIdEEEE.const_reference)
      mutating func push_back(_ __x: std.__1.__CxxTemplateInstNSt3__16vectorIdNS_9allocatorIdEEEE.value_type)
```

However a larger number of test are failing, so I assume checking for canonical type isn’t the expected behavior either:

```
Failed Tests (21):
  Swift(macosx-arm64) :: Interop/Cxx/class/protocol-conformance-typechecker.swift
  Swift(macosx-arm64) :: Interop/Cxx/ergonomics/implicit-computed-properties-module-interface.swift
  Swift(macosx-arm64) :: Interop/Cxx/operators/member-inline.swift
  Swift(macosx-arm64) :: Interop/Cxx/reference/const-ref-parameter.swift
  Swift(macosx-arm64) :: Interop/Cxx/reference/reference-irgen.swift
  Swift(macosx-arm64) :: Interop/Cxx/reference/reference-module-interface.swift
  Swift(macosx-arm64) :: Interop/Cxx/reference/reference-silgen-cxx-objc-ctors+init.swift
  Swift(macosx-arm64) :: Interop/Cxx/reference/reference-silgen.swift
  Swift(macosx-arm64) :: Interop/Cxx/reference/reference.swift
  Swift(macosx-arm64) :: Interop/Cxx/stdlib/overlay/custom-collection-module-interface.swift
  Swift(macosx-arm64) :: Interop/Cxx/stdlib/overlay/custom-collection.swift
  Swift(macosx-arm64) :: Interop/Cxx/stdlib/overlay/custom-iterator-module-interface.swift
  Swift(macosx-arm64) :: Interop/Cxx/stdlib/overlay/custom-sequence-module-interface.swift
  Swift(macosx-arm64) :: Interop/Cxx/stdlib/overlay/custom-sequence-typechecker.swift
  Swift(macosx-arm64) :: Interop/Cxx/stdlib/overlay/custom-sequence.swift
  Swift(macosx-arm64) :: Interop/Cxx/stdlib/use-std-map.swift
  Swift(macosx-arm64) :: Interop/Cxx/templates/defaulted-template-type-parameter.swift
  Swift(macosx-arm64) :: Interop/Cxx/templates/dependent-types.swift
  Swift(macosx-arm64) :: Interop/Cxx/templates/function-template.swift
  Swift(macosx-arm64) :: Interop/Cxx/templates/member-templates-silgen.swift
  Swift(macosx-arm64) :: Interop/Cxx/templates/template-type-parameter-not-in-signature.swift


Testing Time: 21.08s
  Unsupported:  24
  Passed     : 238
  Failed     :  21
```

@zoecarver any ideas/suggestions?